### PR TITLE
Implement statement sequence for the CSE machine

### PR DESCRIPTION
**Pull request by NhatMinh0208**
### Description

This PR implements statement sequences for the CSE machine, as described in #1580. It introduces a new node type, `StatementSequence`, an AST transformer that converts `BlockStatements` and `Programs` without declarations into `StatementSequences` and adds logic to the CSE machine interpreter to handle the new node type. 

### Changes
- [X] Add `StatementSequence`
- [X] Replace `es.Node` with custom `Node` type that includes `StatementSequence` 
- [X] Add an AST transformer that converts `BlockStatements` without declarations into `StatementSequences`
- [X] Add logic to interpreter to handle `StatementSequence`
- [X] Modify astToString to work with `StatementSequence` (for frontend)

### CODE REVIEW

**Comment by NhatMinh0208** [comment]
wow that somehow caused zero regression

**Comment by NhatMinh0208** [comment]
Barring any bugs that are not covered by tests, the CSE machine integration of `StatementSequence` is now complete.

**Review by NhatMinh0208 — COMMENTED** [review]
This is for the frontend - I'll make a separate PR in `frontend` later.

**Review by martin-henz — COMMENTED** [review]
Wow, this turned out to be a major revamp.

Browsed through, and looks fine. Didn't go through it in detail.

File: src/cse-machine/interpreter.ts [hunk 1]
```diff
@@ -311,8 +310,9 @@ export function* generateCSEMachineStateStream(
 
   let command = control.peek()
 
-  // First node will be a Program
```
Comment (RichDom2185) [inline src/cse-machine/interpreter.ts:314]: Is this deleted comment still relevant?

File: src/utils/astToString.ts [hunk 2]
```diff
@@ -1,4 +1,84 @@
...
+      // Line comment
+      state.write('// ' + comment.value.trim() + '\n', comment)
+    } else {
+      // Block comment
+      state.write('/*')
+      reindent(state, comment.value, indent, lineEnd)
+      state.write('*/' + lineEnd)
+    }
+  }
+}
```
Comment (RichDom2185) [inline src/utils/astToString.ts:41]: Minor nit: documentation should use double stars:

```ts
/**
 * Something
 * Something
 */
function f() { /* ... */ }
```

And written above (outside) the function declaration.

File: StatementSequence
Comment (RichDom2185) [inline StatementSequence:1]: What's this empty file for?

**Review by RichDom2185 — COMMENTED** [review]
Overall, LGTM, thanks for the (though out of necessity) refactor to our own type alias!

I'd withhold an approval just because I have clarifications, but in terms of everything else, looks good!

File: StatementSequence
Comment (NhatMinh0208) [reply StatementSequence:1]: Oh that file should not be here - I'll delete it.

File: src/cse-machine/interpreter.ts (hunk 1 above)
Comment (NhatMinh0208) [reply src/cse-machine/interpreter.ts:314]: I'll replace it with a comment that more accurately describes the new code.

**Review by joeng03 — APPROVED** [review]
LGTM!

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**